Assets/Mirror/Runtime/Transport/Websocket/Server.cs

```diff
@@ -272,7 +276,16 @@ public async void Send(int connectionId, byte[] data)
             {
                 try
                 {
-                    await client.SendAsync(new ArraySegment<byte>(data), WebSocketMessageType.Binary, true, cancellation.Token);
+                    messagesToSend.Enqueue(new ArraySegment<byte>(data));
+                    if (!sendingMessage)
+                    {
+                        sendingMessage = true;
+                        while (messagesToSend.Count > 0)
+                        {
+                            await client.SendAsync(messagesToSend.Dequeue(), WebSocketMessageType.Binary, true, cancellation.Token);
+                        }
+                        sendingMessage = false;
+                    }
```

Comment:
@Katori can you explain what happens here? why enqueue something when you dequeue it in the await? why is this better?

Reply:
@vis2k from this thread:  https://forum.unity.com/threads/unity-2017-1-tls-1-2-still-not-working-with-net-4-6.487415/  andreasreich explained that:

> Just for for everyone else following this thread: This is an expected error for attempting interleaved asynchronous write calls on SslStream like BeginWrite and no bug in Unity. Using this advice Bhearus was able to fix the issue in his project.

Suppose 3 sends get called immediately.  The first send will queue and deque i right away, call SendAsync and await.
The second send will add to the queue and exit
the third send will add to he queue and exit
Eventually the first send finishes and continues to send the second and third message.

This queue ensures that only one SendAsync is going on at a time, which causes the exception with TLS

Reply:
doesn't await block the thread until the async finishes?

Reply:
No.  await blocks the *Task*,    the main thread moves on to another Task